mod/common/src/main/java/gjum/minecraft/mapsync/common/net/SyncClient.java

```diff
@@ -284,7 +284,7 @@ void setUpEncryption(ChannelHandlerContext ctx, ClientboundEncryptionRequestPack
 	}
 
 	private static byte[] encrypt(PublicKey key, byte[] data) throws NoSuchPaddingException, NoSuchAlgorithmException, BadPaddingException, IllegalBlockSizeException, InvalidKeyException {
-		Cipher cipher = Cipher.getInstance("RSA/ECB/PKCS1Padding");
+		Cipher cipher = Cipher.getInstance("RSA/NONE/OAEPPadding"); // RSA_PKCS1_PADDING is no longer supported for private decryption
```

Comment:
It may be worth switching to websockets (#43) and letting TLS do all the encryption, though this *does* add complexity in terms of self-signed certs.

Reply:
if you run map sync behind a reverse proxy, (as should be the recommended setup anyway,) self signed certs become less of an issues especially with many featuring lets encrypt integration, or highly popular tooling around that.

Reply:
That's true, but it also adds more complexity to server setup: the client will no-doubt expect to connect to a `wss://` target, but running the server directly would not produce a `wss://` target to connect to, which likely means a higher reliance on docker (or similar), which is not particularly noob-friendly. And this is *MapSync*, we shouldn't expect our users to be highly technically literate. Setting up a MapSync server should be as easy as running those OTT executables that were handed out to people stuck in Wholefoods.

Reply:
I mean that's a somewhat noble goal, I just feel like its unrealistic that many will run the server at all. Who knows though, if that is the case for whatever reason, we could always embed a lets encrypt library and do it all ourselves.